source/components/composing-components.md

```diff
@@ -0,0 +1,260 @@
+    {{component (concat 'basic-card-' user.type) user=user}}
+  {{else}}
+    {{component (concat 'detailed-card-' user.type) user=user}}
+  {{/if}}
+{{/user-list}}
+```
+
+Now we have `{{basic-card-superuser}}` and `{{detailed-card-public}}` rendering in their respective places
+based on the `user.type` equaling 'superuser' and 'public' respectively. Here we used the 'concat' helper
+as a subexpression inside of our `{{component}}` helper. Subexpressions are evaluated before the helper
+and then the helper can use the value, in this case it's just a concatenation of two values. You can
+also experiment with other helpers as subexpressions, like the `if` helper.
+
+Apart from the use of `{{component}}`, we also have the second yielded value for our basic/detailed mode feature,
+which we've added to our yield, i.e. `{{yield user basic}}`.
+
+Something to remember is that the yielded values are ordered sequentially by how they are defined, so you
+access them in the same order as the component exposes them.
+
+The consumer can decide how to name the yielded values. We could have named our yields like so:
+
+```app/users/template.hbs
+{{#user-list users=users as |userModel isBasic|}}
+  {{! something creative here }}
+{{/user-list}}
+```
+
+Just remember to respect the order that they are yielded.
+
+### Actions Up
+
+Now that we know how to send data down, we probably want to manipulate that data via some user interaction,
+like changing a user's avatar. We can accomplish this by using actions.
+
+We'll be looking at a `{{user-profile}}` component that implements a save action, which we
+can yield to the block. This will allow a consumer to use this component without having to know how to save,
+but to be able to trigger a save none the less.
+
+Here's our component's definition:
+
+```app/components/user-profile/component.js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  actions: {
+    saveUser() {
+      var user = this.get('user');
+
+      user.save()
+        .then(() => {
+          this.set('saved', true);
+        })
+        .catch(error => {
+          this.set('saveError', error);
+        });
+    }
+  }
+});
+```
+
+```app/components/user-profile/template.hbs
+{{! most likely we have some markup here }}
+{{yield profile (action "saveUser")}}
+```
+
+So now the consumer will have access to our data and the
+action that we have defined. Lets see how this component could be consumed
+in block form.
+
+```app/templates/user/profile.hbs
+{{#user-profile user=user as |profile saveUser|}}
+  {{user-avatar url=profile.imageUrl onchange=saveUser}}
+{{/user-profile}}
+```
+
+As you can see from this example, the consumer was able to hook into the
+`{{user-profile}}` component's `saveUser` action, which we passed down
+with the `(action "saveUser")` subexpression. This subexpression allows us
+to bundle the action and the target into one value so we don't have to
+specify a `targetObject` property.
```

Comment:
~~subexpression~~, ~~targetObject~~ I don't think they will knoe what that means.

`(action` reads a property off the actions hash of the current context, then creates a closure over that function and any arguments you pass.